lang/nodejs/action.yml

```diff
@@ -1,13 +1,34 @@
 name: 'Publish to NPM'
-description: 'A subcomponent of pulumi-package-publish'
+description: |
+  A subcomponent of pulumi-package-publish
+
+inputs:
+  version:
+    description: The version of the provider being published.
+    required: true
+  node-version:
+    description: The version of Node being used in the provider
+    required: true
 
 runs:
   using: "composite"
   steps:
+    - name: Validate Package Version
+      shell: bash
+      # Ensure that we are publishing only the version that we expect (and that is set in
+      # inputs.version). Otherwise we could publish a version that doesn't match what we
+      # are trying to publish.
+      run: >-
+        ACTUAL="$(cat ${{ github.workspace }}/sdk/nodejs/bin/package.json | jq .version)" &&
+        JS=$(pulumictl convert-version --version "${{ inputs.version }} --language javascript &&
```

Comment:
Can we assume pulumictl is available here? Don't we need to install it?

Reply:
We can assume that it is installed, since we installed it earlier:

https://github.com/pulumi/pulumi-package-publisher/blob/8638d9d2587ba5c8aee60544d8e7a0da126d4c50/action.yml#L78-L81